mteb/tasks/Retrieval/multilingual/BelebeleRetrieval.py

```diff
@@ -0,0 +1,201 @@
+    "fin": ["fin-Latn"],
+    "fra": ["fra-Latn"],
+    "fuv": ["fuv-Latn"],
+    "gaz": ["gaz-Latn"],
+    "grn": ["grn-Latn"],
+    "guj": ["guj-Gujr"],
+    "hat": ["hat-Latn"],
+    "hau": ["hau-Latn"],
+    "heb": ["heb-Hebr"],
+    "hin": ["hin-Deva"],
+    "hrv": ["hrv-Latn"],
+    "hun": ["hun-Latn"],
+    "hye": ["hye-Armn"],
+    "ibo": ["ibo-Latn"],
+    "ilo": ["ilo-Latn"],
+    "ind": ["ind-Latn"],
+    "isl": ["isl-Latn"],
+    "ita": ["ita-Latn"],
+    "jav": ["jav-Latn"],
+    "jpn": ["jpn-Jpan"],
+    "kac": ["kac-Latn"],
+    "kan": ["kan-Knda"],
+    "kat": ["kat-Geor"],
+    "kaz": ["kaz-Cyrl"],
+    "kea": ["kea-Latn"],
+    "khk": ["khk-Cyrl"],
+    "khm": ["khm-Khmr"],
+    "kin": ["kin-Latn"],
+    "kir": ["kir-Cyrl"],
+    "kor": ["kor-Hang"],
+    "lao": ["lao-Laoo"],
+    "lin": ["lin-Latn"],
+    "lit": ["lit-Latn"],
+    "lug": ["lug-Latn"],
+    "luo": ["luo-Latn"],
+    "lvs": ["lvs-Latn"],
+    "mal": ["mal-Mlym"],
+    "mar": ["mar-Deva"],
+    "mkd": ["mkd-Cyrl"],
+    "mlt": ["mlt-Latn"],
+    "mri": ["mri-Latn"],
+    "mya": ["mya-Mymr"],
+    "nld": ["nld-Latn"],
+    "nob": ["nob-Latn"],
+    "npi": ["npi-Deva"],
+    "nso": ["nso-Latn"],
+    "nya": ["nya-Latn"],
+    "ory": ["ory-Orya"],
+    "pan": ["pan-Guru"],
+    "pbt": ["pbt-Arab"],
+    "pes": ["pes-Arab"],
+    "plt": ["plt-Latn"],
+    "pol": ["pol-Latn"],
+    "por": ["por-Latn"],
+    "ron": ["ron-Latn"],
+    "rus": ["rus-Cyrl"],
+    "shn": ["shn-Mymr"],
+    "sin": ["sin-Latn"],
+    "slk": ["slk-Latn"],
+    "slv": ["slv-Latn"],
+    "sna": ["sna-Latn"],
+    "snd": ["snd-Arab"],
+    "som": ["som-Latn"],
+    "sot": ["sot-Latn"],
+    "spa": ["spa-Latn"],
+    "srp": ["srp-Cyrl"],
+    "ssw": ["ssw-Latn"],
+    "sun": ["sun-Latn"],
+    "swe": ["swe-Latn"],
+    "swh": ["swh-Latn"],
+    "tam": ["tam-Taml"],
+    "tel": ["tel-Telu"],
+    "tgk": ["tgk-Cyrl"],
+    "tgl": ["tgl-Latn"],
+    "tha": ["tha-Thai"],
+    "tir": ["tir-Ethi"],
+    "tsn": ["tsn-Latn"],
+    "tso": ["tso-Latn"],
+    "tur": ["tur-Latn"],
+    "ukr": ["ukr-Cyrl"],
+    "urd": ["urd-Arab"],
+    "uzn": ["uzn-Latn"],
+    "vie": ["vie-Latn"],
+    "war": ["war-Latn"],
+    "wol": ["wol-Latn"],
+    "xho": ["xho-Latn"],
+    "yor": ["yor-Latn"],
+    "zho": ["zho-Hans"],
+    "zsm": ["zsm-Latn"],
+    "zul": ["zul-Latn"],
+}
+
+
+class BelebeleRetrieval(MultilingualTask, AbsTaskRetrieval):
+    metadata = TaskMetadata(
+        name="BelebeleRetrieval",
+        dataset={
+            "path": "facebook/belebele",
+            "revision": "75b399394a9803252cfec289d103de462763db7c",
+        },
+        description=(
+            "Belebele is a multiple-choice machine reading comprehension (MRC) dataset spanning 115 distinct languages."
+        ),
+        type="Retrieval",
+        category="s2p",
+        eval_splits=[_EVAL_SPLIT],
+        eval_langs=_LANGS,
+        reference="https://arxiv.org/abs/2308.16884",
+        main_score="ndcg_at_20",
```

Comment:
Yes that would make sense if we wanted to get a correct answer across the 4 possible texts given a passage and a question. However, here we treat the problem as a retrieval of passages based on questions. I put 20 initially because I had also included answer retrieval (https://github.com/embeddings-benchmark/mteb/pull/636#discussion_r1590807508), but now that I changed it maybe it makes more sense to have ndcg_at_10, wdyt?

Reply:
Why would we treat the task of this dataset differently than it is used and introduced in the paper. 


Unless there is a series of evaluations using this dataset otherwise, you should only evaluate the dataset in the same way as the introducing paper and the evaluation method of it. MCQA would work perfectly fine within the AbsRetrieval task.

Reply:
I'm not sure if adapting a dataset to a slightly different task is such a terrible thing 😅

Many reading comprehension datasets are treated that way in the retrieval category:
https://github.com/embeddings-benchmark/mteb/blob/main/mteb/tasks/Retrieval/fra/FQuADRetrieval.py
https://github.com/embeddings-benchmark/mteb/blob/main/mteb/tasks/Retrieval/deu/GermanQuADRetrieval.py

However, if you think it can be easily implemented within the `AbsRetrieval` class, please suggest how and I'll be glad to do it

Reply:
Yeah we already included *QuAD datasets for retrieval and even summarization ones. Imo it's okay to use it as a retrieval dataset as well

Reply:
I don't think it's a bad thing to use a dataset for a slightly different task, but why not the original one? PR #645 should be a good starting point.

Query: the passage (a bit unintuitive)
Corpus: the four questions - positive one marked as positive

Then, essentially you want to see if the highest cosine similarity of the answers, and if it's the right one, the model predicted "accurately". 

On MTEB terms, all you need to do is:
metric: recall_at_1 (or precision at 1 but use recall)

The embedding models might perform worse than the LLM's as described in the original paper but I would expect some performance. Please double check this in the results

@imenelydiaker just double checking that it's okay that I request the evaluation to be according to the original of. the paper and not a similar task. My intuition is that if I read the paper and hear it's in the benchmark I want to see the same kind of evaluation to see how the field evolved.

Reply:
So if I het it right, the change you're suggesting is to change the main score to recall_at_1 instead of ndcg_at_20? In this case, then okay, all metrics will be computed anyway.

Else, the retrieval evaluator is the same for all tasks so I'm not sure we're changing it for this task only.

Reply:
> Query: the passage (a bit unintuitive)
> Corpus: the four questions - positive one marked as positive

I think you meant "the four answers".
Can you clarify how the question itself fits into this? Seems like we only have passages and answers with this setup

Reply:
> > Query: the passage (a bit unintuitive)
> > Corpus: the four questions - positive one marked as positive
> 
> I think you meant "the four answers". Can you clarify how the question itself fits into this? Seems like we only have passages and answers with this setup

Yes, Four answers. I guess prepending the question to the answers might be sufficient.

Reply:
Anyway, if within the benchmark there are more QuAD used as a pure retrieval task, feel free to keep as is. I am removing my request to review changes, I think it was worth the consideration anyway! Thank you both

Reply:
I believe atm. even if we were to use recall_at_1 it would not be eq. to the accuracy in the paper. If that is the case I would prefer to use ndcg_at_10 both to avoid unintended comparisons and to keep it in line with MTEB.

Reply:
Thanks for the insightful review, @Andrian0s!

I understand your point about aligning our evaluation with the one in the paper. However, I didn't think our setup allowed that, which led me to adapt it. Tbh prepending questions to the answers doesn't seem convincing to me, and also even if we used `recall_at_1`, I think it would apply across the entire corpus, not just the four candidate answers as in the paper, so the comparison wouldn’t hold. MCQA could be a cool new task to implement though